contracts/BSCValidatorSet.sol

```diff
@@ -140,26 +157,32 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
   event validatorExitMaintenance(address indexed validator);
 
   /*********************** init **************************/
-  function init() external onlyNotInit{
-    (IbcValidatorSetPackage memory validatorSetPkg, bool valid)= decodeValidatorSetSynPackage(INIT_VALIDATORSET_BYTES);
+  function init() external onlyNotInit {
+    (IbcValidatorSetPackage memory validatorSetPkg, bool valid) = decodeValidatorSetSynPackage(INIT_VALIDATORSET_BYTES);
     require(valid, "failed to parse init validatorSet");
-    for (uint i = 0;i<validatorSetPkg.validatorSet.length;i++) {
+    ValidatorExtra memory _validatorExtra;
+    for (uint i = 0; i < validatorSetPkg.validatorSet.length; i++) {
+      _validatorExtra.voteAddress = validatorSetPkg.voteAddrs[i];
       currentValidatorSet.push(validatorSetPkg.validatorSet[i]);
-      currentValidatorSetMap[validatorSetPkg.validatorSet[i].consensusAddress] = i+1;
+      validatorExtraSet.push(_validatorExtra);
+      currentValidatorSetMap[validatorSetPkg.validatorSet[i].consensusAddress] = i + 1;
     }
     expireTimeSecondGap = EXPIRE_TIME_SECOND_GAP;
+    finalityRewardRatio = FINALITY_REWARD_RATIO;
     alreadyInit = true;
```

Comment:
it should not add any code in `init()`, this is a contract that will be upgraded on BSC,  but `init()` only be executed while deployed.  If we add some code on `init()`,  it will cause tests out of action

Reply:
Fixed